src/metabase/api/table.clj

```diff
@@ -48,6 +48,12 @@
     (t2/hydrate tables :db)
     (filterv mi/can-read? tables)))
 
+(api/defendpoint GET "/uploaded"
+  "Get all `Tables` visible to the current user which were created by uploading a file."
+  []
+  (as-> (t2/select Table, :active true, :is_upload true, {:order-by [[:name :asc]]}) tables
+        (filterv mi/can-read? tables)))
```

Comment:
I removed the hydration done in the regular list because it didn't seem to affect the data returned by the endpoint.

I am somewhat anxious that there was some other reason for this step though, perhaps it prevents N+1 queries when checking permissions?